Library/Homebrew/sorbet/tapioca/compilers/forwardables.rb

```diff
@@ -0,0 +1,79 @@
+# typed: strict
+# frozen_string_literal: true
+
+require_relative "../../../global"
+require "sorbet/tapioca/utils"
+require "utils/ast"
+
+module Tapioca
+  module Compilers
+    class Forwardables < Tapioca::Dsl::Compiler
+      FORWARDABLE_FILENAME = "forwardable.rb"
+      ARRAY_METHODS = T.let(["to_a", "to_ary"].freeze, T::Array[String])
+      HASH_METHODS = T.let(["to_h", "to_hash"].freeze, T::Array[String])
+      STRING_METHODS = T.let(["to_s", "to_str", "to_json"].freeze, T::Array[String])
+
+      ConstantType = type_member { { fixed: Module } }
+
+      sig { override.returns(T::Enumerable[Module]) }
+      def self.gather_constants
+        objects = T.cast(ObjectSpace.each_object(Forwardable), T::Enumerator[Module])
+        objects.map { |obj| named_object(obj) }.uniq.reject do |obj|
+          # Avoid duplicate stubs for forwardables that are defined in vendored gems
+          Object.const_source_location(T.must(obj.name))&.first&.include?("vendor/bundle/ruby")
+        end
+      end
+
+      sig { params(obj: Module).returns(Module) }
+      def self.named_object(obj)
```

Comment:
Tapioca [filters](https://github.com/Shopify/tapioca/blob/3948457/lib/tapioca/dsl/pipeline.rb#L173) out anonymous modules*, including singleton classes, so we use the attached object here, then iterate over both the instance and class methods in the `decorate` method below. I don't love it, but it does the job. (I might make a second pass over this in the future to implement a more elegant solution.)

*This behavior took me a really long time to discover